### PR TITLE
improve tests L2CrossDomainMessenger_Test

### PR DESCRIPTION
**Pull request by AmadiMichael**
improve tests L2CrossDomainMessenger_Test

### CODE REVIEW

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 1]
```diff
@@ -148,17 +149,85 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+    /// signified.
+    function test_relayMessage_callerIsOptimismPortalChecks_reverts() external {
+        // set the target to be the OptimismPortal
+        address target = alice;
+        address sender = address(l1CrossDomainMessenger);
+        address caller = AddressAliasHelper.applyL1ToL2Alias(address(l1CrossDomainMessenger));
+        bytes memory message = hex"1111";
+
+        // cannot send a message as optimism portal but amount does not match msg.value
+        vm.deal(caller, 10 ether);
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol:161]: Same comment as before, let's break this into 3 tests

Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol:161]: fixed

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 2]
```diff
@@ -148,17 +149,92 @@ contract L2CrossDomainMessenger_Test is CommonTest {
         assertEq(l2CrossDomainMessenger.failedMessages(hash), false);
     }
 
-    /// @dev Tests that `relayMessage` reverts if attempting to relay
-    ///      a message sent to an L1 system contract.
+    /// @dev Tests that relayMessage reverts if caller is optimismPortal and the value sent does not match the amount
+    function test_relayMessage_callerIsOptimismPortalAndValueDoesNotMatchAmount_reverts() external {
+        // set the target to be the OptimismPortal
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]: Comment is wrong

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 3]
```diff
@@ -148,17 +149,92 @@ contract L2CrossDomainMessenger_Test is CommonTest {
         assertEq(l2CrossDomainMessenger.failedMessages(hash), false);
     }
 
-    /// @dev Tests that `relayMessage` reverts if attempting to relay
-    ///      a message sent to an L1 system contract.
+    /// @dev Tests that relayMessage reverts if caller is optimismPortal and the value sent does not match the amount
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]: Caller isn't the OptimismPortal here, can rename the test to just be `test_relayMessage_valueDoesNotMatchAmount_reverts`

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 4]
```diff
@@ -148,17 +149,92 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+        vm.deal(caller, 10 ether);
+        vm.prank(caller);
+        vm.expectRevert(stdError.assertionError);
+        l2CrossDomainMessenger.relayMessage{ value: 10 ether }(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, target, 9 ether, 0, message
+        );
+    }
+
+    /// @dev Tests that relayMessage reverts if a failed message is attempted to be replayed via the optimismPortal
+    function test_relayMessage_callerIsOptimismPortalAndFailedMessageReplay_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]: Same as above generally

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 5]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
         assertEq(l2CrossDomainMessenger.failedMessages(hash), false);
     }
 
-    /// @dev Tests that `relayMessage` reverts if attempting to relay
-    ///      a message sent to an L1 system contract.
+    /// @dev Tests that relayMessage reverts if the value sent does not match the amount
+    function test_relayMessage_valueDoesNotMatchAmount_reverts() external {
+        // set the target to be the alice
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]:
```suggestion
        // set the target to be alice
```

Comment (smartcontracts) [reply packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]: Or honestly can just remove this comment

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 6]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+        address target = alice;
+        address sender = address(l1CrossDomainMessenger);
+        address caller = AddressAliasHelper.applyL1ToL2Alias(address(l1CrossDomainMessenger));
+        bytes memory message = hex"1111";
+
+        // cannot send a message where the amount inputted does not match the msg.value
+        vm.deal(caller, 10 ether);
+        vm.prank(caller);
+        vm.expectRevert(stdError.assertionError);
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol:163]: Note to self, get rid of this annoying assertion. I know why we originally did it but I think it's silly.

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 7]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+        l2CrossDomainMessenger.relayMessage{ value: 10 ether }(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, target, 9 ether, 0, message
+        );
+    }
+
+    /// @dev Tests that relayMessage reverts if a failed message is attempted to be replayed and the caller is the other
+    /// messenger
+    function test_relayMessage_failedMessageReplay_reverts() external {
+        // set the target to be the alice
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]: Can remove this comment like above

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 8]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+        vm.prank(caller);
+        vm.expectRevert(stdError.assertionError);
+        l2CrossDomainMessenger.relayMessage(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, target, 0, 0, message
+        );
+    }
+
+    /// @dev Tests that relayMessage reverts if attempting to relay a message
+    ///      sent to an L1 system contract.
     function test_relayMessage_toSystemContract_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]: Can you break this into two tests? One for each address

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 9]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+            address(l2ToL1MessagePasser),
+            0,
+            0,
+            message
+        );
+    }
+
+    /// @dev Tests that the relayMessage function reverts if the message called by non-optimismPortal but not a failed
+    /// message
+    function test_relayMessage_nonFailedMessageCalledByNonOtherMessenger_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]:
```suggestion
    function test_relayMessage_relayingNewMessageByExternalUser_reverts() external {
```

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 10]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
         assertEq(l2CrossDomainMessenger.failedMessages(hash), false);
     }
 
-    /// @dev Tests that `relayMessage` reverts if attempting to relay
-    ///      a message sent to an L1 system contract.
+    /// @dev Tests that relayMessage reverts if the value sent does not match the amount
+    function test_relayMessage_valueDoesNotMatchAmount_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]:
```suggestion
    function test_relayMessage_fromOtherMessengerValueMismatch_reverts() external {
```

File: packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol [hunk 11]
```diff
@@ -148,17 +149,93 @@ contract L2CrossDomainMessenger_Test is CommonTest {
...
+        vm.prank(caller);
+        vm.expectRevert(stdError.assertionError);
+        l2CrossDomainMessenger.relayMessage{ value: 10 ether }(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, target, 9 ether, 0, message
+        );
+    }
+
+    /// @dev Tests that relayMessage reverts if a failed message is attempted to be replayed and the caller is the other
+    /// messenger
+    function test_relayMessage_failedMessageReplay_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L2/L2CrossDomainMessenger.t.sol]:
```suggestion
    function test_relayMessage_fromOtherMessengerFailedMessageReplay_reverts() external {
```

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**